src/mlpack/methods/dbscan/dbscan.hpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file dbscan.hpp
+ * @author Ryan Curtin
+ *
+ * An implementation of the DBSCAN clustering method, which is flexible enough
+ * to support other algorithms for finding nearest neighbors.
+ */
+#ifndef __MLPACK_METHODS_DBSCAN_DBSCAN_HPP
+#define __MLPACK_METHODS_DBSCAN_DBSCAN_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/range_search/range_search.hpp>
+#include "random_point_selection.hpp"
+#include <boost/dynamic_bitset.hpp>
+
+namespace mlpack {
+namespace dbscan {
+
+template<typename RangeSearchType = range::RangeSearch<>,
+         typename PointSelectionPolicy = RandomPointSelection>
+class DBSCAN
+{
+ public:
+  /**
+   * Construct the DBSCAN object with the given parameters.
+   *
+   * @param epsilon Size of range query.
+   * @param minPoints Minimum number of points for each cluster.
+   */
+  DBSCAN(const double epsilon,
```

Comment:
Should we add a parameter description for ``rangeSearch`` and ``pointSelector``?